From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Fri, 24 May 2024 22:52:47 +0100
Subject: [PATCH] xdg: Install livecd variant for kscreenlocker

Which disables autolocking.

It uses /etc/ which is not stateless but it is only for the livecd
so eh.
---
 meson.build                                    |  2 ++
 meson_options.txt                              |  1 +
 xdg/livecd/kscreenlockerrc                     |  6 ++++++
 xdg/livecd/meson.build                         | 16 ++++++++++++++++
 xdg/livecd/plasma-desktop-branding.tmpfiles.in |  2 ++
 xdg/meson.build                                |  2 ++
 6 files changed, 29 insertions(+)
 create mode 100644 xdg/livecd/kscreenlockerrc
 create mode 100644 xdg/livecd/meson.build
 create mode 100644 xdg/livecd/plasma-desktop-branding.tmpfiles.in

diff --git a/meson.build b/meson.build
index fe0c3e2..01bcd31 100644
--- a/meson.build
+++ b/meson.build
@@ -13,6 +13,7 @@ project(
 
 path_prefix = get_option('prefix')
 path_datadir = join_paths(path_prefix, get_option('datadir'))
+path_livecddir = join_paths(path_datadir, 'livecd')
 path_schemadir = join_paths(path_datadir, 'glib-2.0', 'schemas')
 
 # Solusism
@@ -35,6 +36,7 @@ report = [
     '    prefix:                                 @0@'.format(path_prefix),
     '    datadir:                                @0@'.format(path_datadir),
     '    envdir:                                 @0@'.format(path_envdir),
+    '    livecddir:                              @0@'.format(path_livecddir),
 ]
 
 # Output some stuff to validate the build config
diff --git a/meson_options.txt b/meson_options.txt
index 4758439..31e7559 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1 +1,2 @@
+option('tmpfilesdir', description: 'Location for installing tmpfiles.d configs', type: 'string', value : '/usr/lib/tmpfiles.d')
 option('with-env-dir', description: 'Plasma Workspace env directory', type: 'string')
diff --git a/xdg/livecd/kscreenlockerrc b/xdg/livecd/kscreenlockerrc
new file mode 100644
index 0000000..766f018
--- /dev/null
+++ b/xdg/livecd/kscreenlockerrc
@@ -0,0 +1,6 @@
+[Greeter]
+Theme=org.kde.breezedark.desktop
+
+[Daemon]
+Autolock=false
+LockOnResume=false
diff --git a/xdg/livecd/meson.build b/xdg/livecd/meson.build
new file mode 100644
index 0000000..ad82f9b
--- /dev/null
+++ b/xdg/livecd/meson.build
@@ -0,0 +1,16 @@
+livexdg_data = [
+    'kscreenlockerrc',
+]
+
+install_data(
+    livexdg_data,
+    install_dir: path_livecddir,
+)
+
+conf = configuration_data()
+conf.set('LIVECD_DIR', path_livecddir)
+
+configure_file(input : 'plasma-desktop-branding.tmpfiles.in',
+  output : 'plasma-desktop-branding.conf',
+  install_dir: get_option('tmpfilesdir'),
+  configuration : conf)
diff --git a/xdg/livecd/plasma-desktop-branding.tmpfiles.in b/xdg/livecd/plasma-desktop-branding.tmpfiles.in
new file mode 100644
index 0000000..db7c5eb
--- /dev/null
+++ b/xdg/livecd/plasma-desktop-branding.tmpfiles.in
@@ -0,0 +1,2 @@
+# This file will only exist at runtime
+L+ /run/xdg/kscreenlockerrc - - - - ../..@LIVECD_DIR@/kscreenlockerrc
diff --git a/xdg/meson.build b/xdg/meson.build
index c5b7e0e..0253994 100644
--- a/xdg/meson.build
+++ b/xdg/meson.build
@@ -26,3 +26,5 @@ install_data(
     xdg_data,
     install_dir: xdg_dir,
 )
+
+subdir('livecd')
